py/modsys.c

```diff
@@ -166,16 +166,6 @@ STATIC mp_obj_t mp_sys_getsizeof(mp_obj_t obj) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(mp_sys_getsizeof_obj, mp_sys_getsizeof);
 #endif
 
-#if MICROPY_PY_SYS_ATEXIT
-// atexit(callback): Callback is called when sys.exit is called.
-STATIC mp_obj_t mp_sys_atexit(mp_obj_t obj) {
-    mp_obj_t old = MP_STATE_VM(sys_exitfunc);
-    MP_STATE_VM(sys_exitfunc) = obj;
-    return old;
-}
-STATIC MP_DEFINE_CONST_FUN_OBJ_1(mp_sys_atexit_obj, mp_sys_atexit);
-#endif
-
```

Comment:
I don't think we want to _remove_ this upstream code, just ensure that we don't define MICROPY_PY_SYS_ATEXIT (1). Deleting the lines may cause additional merge conflicts, in the case that micropython revises the lines we deleted.

Reply:
I don't think there will be a need to revise the code based on upstream changes as the new `atexit` implementation is completely different from what micropython provides, which is not CPython compatible and limited in functionality.

Reply:
Good call, I'll add that back.